Better Directory Input & Game Object based Lip Syncing

deviandice opened the pull request:
**1. Enhanced Directory Wizard**
The existing wizard has a **_significant_** limitation: It only reads the children of a specified directory without replicating that directory's structure in the output. 
This oversight forces users to manually sort their baked data to match the hierarchy of the original audio clips, which is inefficient, prone to errors, and if your line count is high enough may lead to death from dehydration because of how long sorting them would take. My alternative wizard addresses this issue by cloning the directory hierarchy from input to output, ensuring that the structure of subfolders and their contents is preserved. 

**2. Game Object-based Lip Syncing for 2D Rigs**
I noticed a very noticeable gap in the current toolset for 2D lipsyncing, particularly for rigs that aren't sprite sheet-based (Believe it or not that's every single rig I use). uLipSyncTexture falls short in this case. One of Unity's only features that actually "just works" is the PSB import pipeline, which automatically slices sprites based on their boundaries, resulting in varying origins and sizes that aren't easily swappable. It also mirrors the layer structure when dragging the imported PSB into the editor. Therefore, all of your phonemes will already exist as individual game objects. So I created uLipSyncGameObject which is functionally similar to uLipSyncTexture, but works by just enabling and disabling the objects instead. 

I love this tool, so I hope these are useful additions.

hecomi commented:
Thank you for the fantastic PR! I aim to review and integrate it as soon as possible. However, I might not have time in the immediate future, so it could take up to a month. I appreciate your patience. Additionally, since there are two updates included, it would be smoother to review if you could separate them into individual PRs.

deviandice commented:
Updating my Pull Request.
In a backwards-compatible and non-destructive refactor, I converted baked data to use sparse phoneme lists. This reduced 1000 lipsync files containing 9 phonemes of varying lengths from 100 MB to 38 MB, roughly a 60% decrease in disk storage for that number of phonemes. This was requested [here](https://github.com/hecomi/uLipSync/issues/68). I have added a context menu option to convert the baked data to sparse and modified the base baked command to automatically bake new audio with sparse frames. This does not break any functionality as far as I can see. 

Here is an example of the size reduction before and after converting to sparse.
![image](https://github.com/user-attachments/assets/e3ed692b-5e9f-478b-a234-52b189ba72c0)
